examples/user_guide/04_Machine_Learning.ipynb

```diff
@@ -4,8 +4,485 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
+    "# Machine Learning\n",
+    "\n",
     "With the data preparation complete, this user guide will demonstrate how you can configure a [`scikit-learn`](http://scikit-learn.org/stable/index.html) pipeline. In the next user guide, [Data_Visualization](05_Data_Visualization) you will learn how to visualize the output of this pipeline and diagnose as well as ensure that the inputs to the pipeline have the expected structure."
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Recap: Loading data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import intake\n",
+    "\n",
+    "cat = intake.open_catalog('../catalog.yml')\n",
+    "l5_da = cat.l5.read_chunked()\n",
+    "l5_da"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Subsetting data\n",
+    "\n",
+    "We'll use a subset of these data in the following examples. There are many ways to [subset data in xarray](http://xarray.pydata.org/en/stable/indexing.html). Here we select a central 2000x2000 pixel piece using index selection. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "l5_da = l5_da[:, 3000:5000, 3000:5000]\n",
+    "l5_da"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import hvplot.xarray"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "l5_da.hvplot(kind='image', x='x', y='y', groupby='band', datashade=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Reshaping Data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We'll reshape the image to be how dask-ml / scikit-learn expect it: `(n_samples, n_features)` where n_features is the number of bands in this case. Then we'll persist that in memory. We still have a small dataset at this point. The large dataset, which dask helps us manage, is the intermediate `n_samples x n_samples` array that spectral clustering operates on. For our 2,000 x 2,000 pixel subset, that's 4 million pixels"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Numpy\n",
+    "\n",
+    "Data can be reshaped at the lowest level using numpy. For instance by getting the values from the `xarray.DataArray`, and using flatten and transpose to get the right shape."
```

Comment:
Might be worth explicitly pointing out that there are 6 bands in the shape shown...